src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,152 @@
+package duke;
+
+import duke.commands.Commands;
+import duke.dukehelper.Parser;
+import duke.dukehelper.Storage;
+import duke.dukehelper.TaskList;
+import duke.dukehelper.Ui;
+import duke.exception.DukeException;
+import duke.helper.DateTimeHelper;
+import duke.task.Task;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+
+public class Duke {
+    private Ui ui;
+    private Storage storage;
+    private TaskList tasks;
+    private Parser parser;
+    /**
+     * Constructor
+     */
+    public Duke() {
+        this.ui = new Ui();
+        this.storage = new Storage("data/save_file.txt");
+        this.tasks = new TaskList();
+        this.parser = new Parser();
+    }
+    private int getNumTasks() {
+        return this.tasks.getTaskList().size();
+    }
+
+    /**
+     * Distributes each command based on its type, then return a string that will be shown to the user.
+     * @param commandType
+     * @param tokens
+     * @param isLoaded
+     * @return message to user
+     * @throws DukeException
+     */
+    protected String furtherProcessing(Commands commandType, String[] tokens, boolean isLoaded) throws DukeException {
```

Comment:
Why do you choose to put some processing methods in this class rather than in the `Parser` class or at least in another class? Do you think it would demonstrate better encapsulation if Duke was not in charge of the processing of user inputs?